docs/CACHING.md

```diff
@@ -0,0 +1,23 @@
+## Caching
+Plio uses the Redis cache backend powered by the [Django Redis](https://github.com/jazzband/django-redis) package.
+
+This guide aims to provide details on how Plio is using caching and pre-requisites for someone contributing to the code.
+
+### Caching workflow
+Plio's caching mechanism can be explained in three simple steps:
+1. Receives an API request
+2. Searches for the requested data in the cache
+3. If not present, query the database and save the response in cache for future requests
+
+![Overview of caching](images/caching-workflow.png)
```

Comment:
One thing is not clear:
- what happens if the value is present in the cache but has been updated since it was put in the cache?

Reply:
Okay, makes sense. I'll update it or maybe create a separate one to explain the flow better.